apps/studio/prisma/scripts/createSite.ts

```diff
@@ -0,0 +1,236 @@
+    url: "/contact-us",
+  },
+  {
+    title: "Something else",
+    url: "/something-else",
+  },
+  {
+    title: "Resources",
+    url: "/resources",
+  },
+]
+
+export const createSite = async (siteName: string) => {
+  const siteId = await db.transaction().execute(async (tx) => {
+    const { id: siteId } = await tx
+      .insertInto("Site")
+      .values({
+        name: siteName,
+        theme: jsonb({
+          colors: {
+            brand: {
+              canvas: {
+                alt: "#bfcfd7",
+                default: "#e6ecef",
+                inverse: "#00405f",
+                backdrop: "#80a0af",
+              },
+              interaction: {
+                hover: "#002e44",
+                default: "#00405f",
+                pressed: "#00283b",
+              },
+            },
+          },
+        }),
+        config: jsonb({
+          theme: "isomer-next",
+          siteName: "MTI",
+          logoUrl: "https://www.isomer.gov.sg/images/isomer-logo.svg",
+          search: undefined,
+          isGovernment: true,
+        }),
+      })
+      .onConflict((oc) =>
+        oc
+          .column("name")
+          .doUpdateSet((eb) => ({ name: eb.ref("excluded.name") })),
+      )
+      .returning("id")
+      .executeTakeFirstOrThrow()
+
+    await tx
+      .insertInto("Footer")
+      .values({
+        siteId,
+        content: jsonb({
+          contactUsLink: "/contact-us",
+          feedbackFormLink: "https://www.form.gov.sg",
+          privacyStatementLink: "/privacy",
+          termsOfUseLink: "/terms-of-use",
+          siteNavItems: FOOTER_ITEMS,
+        }),
+      })
+      .onConflict((oc) =>
+        oc
+          .column("siteId")
+          .doUpdateSet((eb) => ({ siteId: eb.ref("excluded.siteId") })),
+      )
+      .execute()
+
+    await tx
+      .insertInto("Navbar")
+      .values({
+        siteId,
+        content: jsonb(NAV_BAR_ITEMS),
+      })
+      .onConflict((oc) =>
+        oc
+          .column("siteId")
+          .doUpdateSet((eb) => ({ siteId: eb.ref("excluded.siteId") })),
+      )
+      .execute()
+
+    const { id: blobId } = await tx
+      .insertInto("Blob")
+      .values({ content: jsonb(PAGE_BLOB) })
+      .returning("id")
+      .executeTakeFirstOrThrow()
+
+    await tx
+      .insertInto("Resource")
+      .values({
+        draftBlobId: String(blobId),
+        permalink: "",
+        siteId,
+        type: "RootPage",
+        title: "Home",
+      })
+
+      .onConflict((oc) =>
+        oc.column("draftBlobId").doUpdateSet((eb) => ({
+          draftBlobId: eb.ref("excluded.draftBlobId"),
+        })),
+      )
+      .executeTakeFirstOrThrow()
+
+    console.log(`Added site ${siteName} with id ${siteId} to database`)
+    return siteId
+  })
+
+  await addUsersToSite(
+    siteId,
+    [...ISOMER_ADMINS, ...ISOMER_MIGRATORS].map(
+      (email) => `${email}@open.gov.sg`,
+    ),
+  )
+
+  await updateUserPermissions(siteId, RoleType.Admin)
```

Comment:
would it be a better idea to not have these functions be included here?

Reply:
actually i want to have them here just because it makes it easier for us to run the script! i put fake values so that it'll fail and provide feedback, which minimises the risk of running a "false" migration (+ it's wrapped in `tx` also)

lmk if you don't think this is sufficient!